### PR TITLE
[v7] Change cog resource to pandas

### PR DESCRIPTION
**Pull request by haakonvt**
## TODO:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).

### CODE REVIEW

File: cognite/client/data_classes/_base.py [hunk 1]
```diff
@@ -140,30 +140,36 @@ def _load(
...
 
         Args:
-            expand (Sequence[str]): List of row keys to expand, only works if the value is a Dict. Will expand metadata by default.
-            ignore (list[str] | None): List of row keys to not include when converting to a data frame.
-            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata into separate rows.
+            metadata_prefix (str): Prefix to use for the metadata rows, if expanded.
+            ignore (list[str] | None): List of row keys to skip when converting to a data frame. Is applied before expansions.
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`). Does not affect custom data like metadata.
+            convert_timestamps (bool): Convert known attributes storing CDF timestamps (milliseconds since epoch) to datetime. Does not affect custom data like metadata.
```
Comment (doctrino) [inline cognite/client/data_classes/_base.py:157]: Could this be a sequence of strings? Then, have a default with the known attributes, such that the user could just also specify the metadata field they want converted?

File: cognite/client/utils/_time.py [hunk 2]
```diff
@@ -164,44 +165,54 @@ def timestamp_to_ms(timestamp: int | float | str | datetime) -> int:
...
-    return new_item
+def _convert_time_attrs_in_dict(item: dict, to_isoformat: bool) -> dict:
+    for k in TIME_ATTRIBUTES.intersection(item):
+        try:
+            if to_isoformat:
+                item[k] = ms_to_datetime(item[k]).isoformat(sep=" ", timespec="milliseconds")
+            else:
+                item[k] = ms_to_datetime(item[k])
+        except ValueError:
+            pass
```
Comment (doctrino) [inline cognite/client/utils/_time.py:197]:
```suggestion
        with suppress(ValueError):
            if to_isoformat:
                item[k] = ms_to_datetime(item[k]).isoformat(sep=" ", timespec="milliseconds")
            else:
                item[k] = ms_to_datetime(item[k])
```

And remember to `from contextlib import suppress`

File: cognite/client/data_classes/_base.py (hunk 1 above)
Comment (doctrino) [reply cognite/client/data_classes/_base.py:157]: Very uncertain about this suggestion as it increases complexity.

**Review by doctrino — COMMENTED** [review]
Very nice fix.

File: cognite/client/data_classes/_base.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/data_classes/_base.py:157]: Unsure! Could be added later without breaking, so I'll leave it for now.

File: cognite/client/utils/_time.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/utils/_time.py:197]: I went with try-except as `suppress` has quite the larger overhead (with enter and exit methods) - and we call this in a loop. *But I prefer suppress* 😎

**Pull request merged**